quil-rs/src/program/graph.rs

```diff
@@ -516,22 +519,19 @@ fn terminate_working_block(
         return Ok(());
     }
 
-    let block = InstructionBlock::build(working_instructions.to_vec(), terminator, program)?;
+    // This leaves working_instructions and working_label in their respective "empty" states.
+    let block = InstructionBlock::build(std::mem::take(working_instructions), terminator, program)?;
```

Comment:
Rather than cloning through `to_vec` and using `.drain(..)` below to empty the `Vec`, using `mem::take` takes ownership of the `Vec` and leaves an empty one in its place.

See the note on `working_label` for why this seems fine to do.